### DISCUSSION TITLE
How to add a webpart in pnp page and set the advanced sppkg file from App catalog[FEATURE]

### DISCUSSION THREAD

**Discussion by LeoZxy** [pnp/powershell#4065] [General]
How should I use power shell to add a webpart and choose the sppkg file in advanced section, I can get the sppkg file id.

<img width="610" alt="Screenshot 2024-07-03 at 10 03 43" src="https://github.com/pnp/powershell/assets/96618210/8c799ab6-ba3a-4023-a666-4a9c11996040">

**Comment by rockraider213** [comment]
hey, have a look here: 
https://www.sharepointdiary.com/2021/03/add-web-part-to-modern-page-in-sharepoint-online-using-powershell.html

you can use Add-PnPPageWebPart to do it.
